tests/test_apply.py

```diff
@@ -59,379 +59,255 @@
-        dev = device(2)
-        state = np.array([[0, 123.432], [-0.432, 023.4]])
-
-        with pytest.raises(ValueError, match=r"Unitary matrix must be of shape"):
-            dev.apply("QubitUnitary", [0, 1], [state])
-
-    @pytest.mark.parametrize("name, mat", three_qubit)
-    def test_three_qubit_no_parameters(self, init_state, device, name, mat, tol):
-        dev = device(3)
-        state = init_state(3)
+        with pytest.raises(
+            ValueError,
+            match=r"State vector must be of length 2\*\*wires."
+        ):
+            p = np.array([1, 0, 1, 1, 0]) / np.sqrt(3)
+            dev.reset()
+            dev.apply([qml.QubitStateVector(p, wires=[0, 1])])
 
-        dev.apply("QubitStateVector", [0, 1, 2], [state])
-        dev.apply("QubitUnitary", [0, 1, 2], [mat])
-        dev._obs_queue = []
-        dev.pre_measure()
+        with pytest.raises(
+            qml.DeviceError,
+            match="Operation QubitStateVector cannot be used after other Operations have already been applied "
+                                  "on a qulacs.simulator device."
+        ):
+            dev.reset()
+            dev.apply([
+                qml.RZ(0.5, wires=[0]),
+                qml.QubitStateVector(np.array([0, 1, 0, 0]), wires=[0, 1])
+            ])
 
-        res = np.fromiter(dev.probabilities().values(), dtype=np.float64)
-        expected = np.abs(mat @ state) ** 2
-        assert np.allclose(res, expected, **tol)
+    def test_apply_errors_basis_state(self):
+        """Test that apply fails for incorrect basis state preparation."""
+        dev = QulacsDevice(1)
 
-    @pytest.mark.parametrize("theta", [0.5432, -0.232])
-    @pytest.mark.parametrize("name,func", two_qubit_param)
-    def test_single_qubit_parameters(self, init_state, device, name, func, theta, tol):
-        """Test PauliX application"""
-        dev = device(2)
-        state = init_state(2)
+        with pytest.raises(
+            ValueError,
+            match="BasisState parameter must consist of 0 or 1 integers."
+        ):
+            dev.apply([qml.BasisState(np.array([-0.2, 4.2]), wires=[0, 1])])
 
-        dev.apply("QubitStateVector", [0, 1], [state])
-        dev.apply(name, [0, 1], [theta])
-        dev._obs_queue = []
-        dev.pre_measure()
+        with pytest.raises(
+            ValueError,
+            match="BasisState parameter and wires must be of equal length."
+        ):
+            dev.apply([qml.BasisState(np.array([0, 1]), wires=[0])])
 
-        res = np.fromiter(dev.probabilities().values(), dtype=np.float64)
-        expected = np.abs(func(theta) @ state) ** 2
-        assert np.allclose(res, expected, **tol)
+        with pytest.raises(
+            qml.DeviceError,
+            match="Operation BasisState cannot be used after other Operations have already been applied "
+                                  "on a qulacs.simulator device."
+        ):
+            dev.reset()
```

Comment:
worth moving this one out of the context so that it's apparent that `dev.apply` raises the error